File: Lab_6/biblioteca.py
# biblioteca.py
from libro import Libro

class Biblioteca:
    def __init__(self):
        self.libros = []  # Colección de libros

    def agregar_libro(self, libro):
        self.libros.append(libro)

    def buscar_libro_por_isbn(self, isbn):
        for libro in self.libros:
            if libro.isbn == isbn:
                return libro
        return None

    def prestar_libro(self, isbn):
        libro = self.buscar_libro_por_isbn(isbn)
        if libro and libro.estado == "disponible":
            libro.prestar()
            return True
        return False

    def devolver_libro(self, isbn):
        libro = self.buscar_libro_por_isbn(isbn)
        if libro and libro.estado == "prestado":
            libro.devolver()
            return True
        return False

    def mostrar_libros(self):
        for libro in self.libros:
            print(libro)

File: Lab_6/Biblioteca.java
// Biblioteca.java
import java.util.ArrayList;

public class Biblioteca {
    private ArrayList<Libro> libros;

    public Biblioteca() {
        libros = new ArrayList<>();
    }

    public void agregarLibro(Libro libro) {
        libros.add(libro);
    }

    public Libro buscarLibroPorISBN(String isbn) {
        for (Libro libro : libros) {
            if (libro.getIsbn().equals(isbn)) {
                return libro;
            }
        }
        return null;
    }

    public boolean prestarLibro(String isbn) {
        Libro libro = buscarLibroPorISBN(isbn);
        if (libro != null && libro.getEstado().equals("disponible")) {
            libro.prestar();
            return true;
        }
        return false;
    }

    public boolean devolverLibro(String isbn) {
        Libro libro = buscarLibroPorISBN(isbn);
        if (libro != null && libro.getEstado().equals("prestado")) {
            libro.devolver();
            return true;
        }
        return false;
    }

    public void mostrarLibros() {
        for (Libro libro : libros) {
            System.out.println(libro);
        }
    }
}
